src/main/cpp/src/cast_string_to_float.cu

```diff
@@ -241,15 +241,24 @@ class string_to_float {
                                           (_warp_lane == 1 && (_c == 'A' || _c == 'a')) ||
                                           (_warp_lane == 2 && (_c == 'N' || _c == 'n')));
     if (nan_mask == 0x7) {
-      // if we start with 'nan', then even if we have other garbage character, this is a null row.
-      //
-      // if we're in ansi mode and this is not -precisely- nan, report that so that we can throw
-      // an exception later.
-      if (_len != 3) {
-        _valid  = false;
-        _except = _len != 3;
-      }
-      return true;
+      // if we start with 'nan', then even if we have other garbage character(excluding
+      // whitespaces), this is a null row. but for e.g. : "nan   " cases. spark will treat the as
+      // "nan", when the trailing characters are whitespaces, it is still a valid string. if we're
+      // in ansi mode and this is not -precisely- nan, report that so that we can throw an exception
+      // later.
+
+      // move forward the curren position by 3
+      _bpos += 3;
+      _c = __shfl_down_sync(0xffffffff, _c, 3);
+
+      // remove the trailing whitespaces, if there exits
+      remove_leading_whitespace();
```

Comment:
How about renaming the function to a more general name like just `remove_whitespace` since we are removing trailing whitespaces but not leading whitespaces here.

Reply:
This isn't what the function does though. I know exactly what will happen if I call `remove_leading_whitespace("  the quick brown fox");`, but what is the expected output of `remove_whitespace("  the quick brown fox");`? It could easily be expected that it means `thequickbrownfox`, which this function isn't doing for us. The internal implementation that we are moving a pointer around is not immediately obvious, but each time we call that, we are removing whitespace from the start of the current string.

Reply:
Yes, this remove_leading_whitespace() removes the leading whitespace of the remaining string to process. The name make sense this way.